### PR TITLE
Labster Course API.

### PR DESCRIPTION
**Pull request by polesye**
@auraz please review.
doc: https://github.com/Livit/Livit.Learn.LTI/wiki/Labser-Course-API

@TODO:
- Sync xblocks
- Tests

### CODE REVIEW

File: cms/djangoapps/labster/course/urls.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"""
+Labster Course URI specification.
+
+Patterns here should simply point to version-specific patterns.
+"""
+from django.conf.urls import patterns, url, include
+
+
+urlpatterns = patterns(
+    '',
```
Comment (auraz) [inline cms/djangoapps/labster/course/urls.py]: why we need empty string here?

File: cms/djangoapps/labster/course/utils.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+def set_staff(course_key, emails):
+    """
+    Sets course staff.
+    """
+    remove_all_instructors(course_key)
+    for email in emails:
+        try:
+            user = User.objects.get(email=email)
+            add_instructor(course_key, user, user)
```
Comment (auraz) [inline cms/djangoapps/labster/course/utils.py:31]: can we move add_instructor out of try/except?

File: cms/djangoapps/labster/course/utils.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    Returns Simulation id extracted from the passed URI.
+    """
+    return urlparse(uri).path.strip('/').split('/')[-1]
+
+
+def get_parent_xblock(xblock, child_for='sequential'):
+    """
+    Find a parent for the xblock.
+    """
+    while xblock:
```
Comment (auraz) [inline cms/djangoapps/labster/course/utils.py:58]: how slow it is? is this uses xblock cache? edX does not have similar function?

Comment (polesye) [inline cms/djangoapps/labster/course/utils.py:58]: > how slow it is? is this uses xblock cache? edX does not have similar function?

Yes, it's. We've used similar code in Student Notes.

File: cms/djangoapps/labster/course/utils.py (hunk 2 above)
Comment (polesye) [inline cms/djangoapps/labster/course/utils.py:31]: > can we move add_instructor out of try/except?

Yes, we can. But I don't see the reason to do so.

**Comment by polesye** [comment]
@den99f please also review.

File: cms/djangoapps/labster/course/urls.py (hunk 1 above)
Comment (polesye) [inline cms/djangoapps/labster/course/urls.py]: > why we need empty string here?

fixed.

File: cms/djangoapps/labster/course/views.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+        else:
+            super(XBlockInfo, self).__setattr__(name, value)
+
+
+def get_xblock_info(xblock, cache, is_hidden=True, child=None):
+    """
+    Returns information (`is_hidden`(bool), `children`(list)) about the xblock.
+    """
+    info = XBlockInfo(cache.get(xblock))
+    info.is_hidden = is_hidden
```
Comment (auraz) [inline cms/djangoapps/labster/course/views.py]: please rewrite with info.set_visibility method.

**Comment by auraz** [comment]
minor comments and :+1:

**Comment by denys-zarubin** [comment]
:+1:

**Pull request merged**